commons/Common.ml

```diff
@@ -1203,13 +1203,31 @@ let cat file =
     close_in chan;
     List.rev !acc
 
-let read_file2 file =
-  let ic = open_in_bin file  in
-  let size = in_channel_length ic in
-  let buf = Bytes.create size in
-  really_input ic buf 0 size;
-  close_in ic;
-  buf |> Bytes.to_string
+(*
+   This implementation works even with Linux files like /dev/fd/63
+   created by bash's process substitution e.g.
```

Comment:
Maybe you should add use a more elaborated comment like this one you made in the PR description:
> In bash, <(echo contents) is replaced by something like /dev/fd/63 which is a special file of apparent size 0 (as reported by Unix.stat) but contains data (here, "contents\n"). So we can't use Unix.stat or in_channel_length to obtain the size of such file.